src/Serilog.Sinks.MongoDB/Sinks/MongoDB/MongoDBJsonFormatter.cs

```diff
@@ -56,6 +57,15 @@ public class MongoDBJsonFormatter : JsonFormatter
             };
         }
 
+        /// <summary>
+        /// Writes out a dictionary
+        /// </summary>
+        protected override void WriteDictionary(IReadOnlyDictionary<ScalarValue, LogEventPropertyValue> elements, TextWriter output)
+        {
+            elements = FixDictionary(elements);
```

Comment:
Instead of converting the dictionary, wouldn't copying and modifying the implementation of the base class method be more efficient, here?

Reply:
It would be more efficient to do it that way. What made me hesitant to do it that way was the fact that WriteLiteral, and the literalWriters are private in the base class, so that looked like a lot of re-implementation to me.

Am I over-complicating that? Or is that what you were thinking, just copy everything? I didn't like so much code copying and not using the base class, but it would certainly be simpler.

Reply:
Though it's marked `Obsolete`, `WriteJsonProperty()` on the base class should cover a lot of the duplication (it's `protected`)

Reply:
(Or even better, we'd copy the whole `JsonFormatter` type into this project and stop using the obsolete one in Serilog)